src/xrootiface.py

```diff
@@ -86,119 +103,129 @@ def init(inconfig, inlog):
   else:
     homepath = ''
 
-def stat(endpoint, filename, ruid, rgid):
-  '''Stat a file via xroot on behalf of the given uid,gid, and returns (size, mtime). Uses the default xroot API.'''
-  filename = _getfilename(filename)
+
+def stat(endpoint, filepath, userid):
+  '''Stat a file via xroot on behalf of the given userid, and returns (size, mtime). Uses the default xroot API.'''
+  filepath = _getfilepath(filepath)
   tstart = time.clock()
-  rc, statInfo = _getxrdfor(endpoint).stat(filename + _eosargs(ruid, rgid))
+  rc, statInfo = _getxrdfor(endpoint).stat(filepath + _eosargs(userid))
   tend = time.clock()
-  log.info('msg="Invoked stat" filename="%s" elapsedTimems="%.1f"' % (filename, (tend-tstart)*1000))
+  log.info('msg="Invoked stat" filepath="%s" elapsedTimems="%.1f"' % (filepath, (tend-tstart)*1000))
   if statInfo is None:
     raise IOError(rc.message.strip('\n'))
   return {'size': statInfo.size, 'mtime': statInfo.modtime}
 
-def statx(endpoint, filename, ruid, rgid):
-  '''Get extended stat info (inode, ouid, ogid, size, mtime) via an xroot opaque query on behalf of the given uid,gid'''
-  filename = _getfilename(filename)
+
+def statx(endpoint, filepath, userid):
+  '''Get extended stat info (inode, filepath, userid, size, mtime) via an xroot opaque query on behalf of the given userid'''
   tstart = time.clock()
-  rc, rawinfo = _getxrdfor(endpoint).query(QueryCode.OPAQUEFILE, filename + _eosargs(ruid, rgid) + '&mgm.pcmd=stat')
+  rc, info = _getxrdfor(endpoint).query(QueryCode.OPAQUEFILE, _getfilepath(filepath) + _eosargs(userid) + '&mgm.pcmd=stat')
   tend = time.clock()
-  log.info('msg="Invoked stat" filename="%s" elapsedTimems="%.1f"' % (filename, (tend-tstart)*1000))
+  log.info('msg="Invoked stat" filepath="%s" elapsedTimems="%.1f"' % (_getfilepath(filepath), (tend-tstart)*1000))
   if '[SUCCESS]' not in str(rc):
     raise IOError(str(rc).strip('\n'))
-  rawinfo = str(rawinfo)
-  if 'retc=' in rawinfo:
-    raise IOError(rawinfo.strip('\n'))
-  statxdata = rawinfo.split()
-  return {'inode': statxdata[2],
-          'ouid': statxdata[5],
-          'ogid': statxdata[6],
+  info = str(info)
+  if 'retc=' in info:
```

Comment:
What is this retc= and why will it raise an error of we don't have it?

Reply:
Another short for return code. That's how the xrootd python bindings work...

Reply:
Ok, so if we have a return code in the info, something's wrong?

Reply:
yes. Again this is eos specific (and not the object of this PR).